source/agent/quic/e2e-test/patches/samplertcservice.js

```diff
@@ -0,0 +1,569 @@
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+// Copyright (C) <2018> Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+// This file is borrowed from lynckia/licode with some modifications.
+
+/*global require, __dirname, console, process*/
+'use strict';
+
+var express = require('express'),
+  spdy = require('spdy'),
+  bodyParser = require('body-parser'),
+  errorhandler = require('errorhandler'),
+  morgan = require('morgan'),
+  fs = require('fs'),
+  https = require('https'),
+  icsREST = require('./rest');
+
+var app = express();
+
+// app.configure ya no existe
+app.use(errorhandler());
+app.use(morgan('dev'));
+app.use(express.static(__dirname + '/public'));
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({
+  extended: true
+}));
+app.disable('x-powered-by');
+
+app.use(function(req, res, next) {
+  res.header('Access-Control-Allow-Origin', '*');
+  res.header('Access-Control-Allow-Methods', 'POST, GET, PUT, PATCH, OPTIONS, DELETE');
+  res.header('Access-Control-Allow-Headers', 'origin, content-type');
+  res.header('Strict-Transport-Security', 'max-age=1024000; includeSubDomain');
+  res.header('X-Content-Type-Options', 'nosniff');
+  if (req.method == 'OPTIONS') {
+    res.send(200);
+  } else {
+    next();
+  }
+});
+
+icsREST.API.init('_service_ID_', '_service_KEY_', 'https://localhost:3000/', false);
+
+var sampleRoom;
+var pageOption = { page: 1, per_page: 100 };
+(function initSampleRoom () {
+  icsREST.API.getRooms(pageOption, function(rooms) {
+    console.log(rooms.length + ' rooms in this service.');
+    for (var i = 0; i < rooms.length; i++) {
+      if (sampleRoom === undefined && rooms[i].name === 'sampleRoom') {
+        sampleRoom = rooms[i]._id;
+        console.log('sampleRoom Id:', sampleRoom);
+      }
+      if (sampleRoom !== undefined) {
+        break;
+      }
+    }
+    var tryCreate = function(room, callback) {
+      var options = {
+        views: [],
+        transcoding: false
```

Comment:
I see you put owt.js and rest-sample.js in "deps", and put samplertcservice.js in "patches".
If you only change the "options" here, maybe you can directly modify the original "samplertcservice.js" and add a condition to check the mode like "process.env.GITHUB_ACTION" in other files.

And another question is what's the version of "deps" and "patches" files from js repo? Will they be updated in the future?

Reply:
Yes, only `options` is changed. Since it's different from the original version, I put it in patches instead of deps. As sampleservice.js is not developed in this repo, it could be used by other GitHub workflows, changing behaviors depends on `process.env.GITHUB_ACTION` may impact other workflows. Perhaps we don't need the entire JavaScript conference sample. Only a REST service to get tokens is enough. I'll try to optimize it in a separate PR.

The latest commit of JavaScript SDK and sample are used for this workflow.

Reply:
Removed unnecessary methods from this file, and renamed directory to service.

Reply:
You said that the latest commit of JavaScript SDK and sample are used.
But JavaScript SDK will have newer commit in upstream, will you update files like "deps/owt.js" in the future?

Reply:
I removed deps/owt.js from this PR. It will be created at runtime.

The second step "Checkout JavaScript SDK and sample" checks out the latest JavaScript SDK. Another step "Build JavaScript SDK and sample release package" runs grunt to build the SDK. Then it'll be copied to deps by `source/agent/quic/e2e-test/scripts/prepare.py`. So no need to manually update this file.